### PR TITLE
Links enhancements and fixes

### LINKED ISSUE #2774: CTRL+K no longer allows immediate pasting of links
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

It used to be the case when you mark text and press CTRL+K, you can immediately paste the link. This is no longer the case, see video:

https://github.com/user-attachments/assets/069ea628-971d-46dc-83a4-aef24ed1309b

### Screenshots

_No response_

### Steps To Reproduce

1. Select text
2. Press CTRL+K

### Expected behavior

I can immediately paste the link

### Logs

_No response_

### Device information

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Soxasora**
## Description

Fixes #2774 
Introduces the `$moveSelection` utility that enables precise range selections and caret placement, we're using it in markdown mode to select either the alt text or the URL after creating a markdown link;
Adds support for `ItemMentionNode` (toggle off);
Better separation of concerns by alleviating transformer bridge of link responsibilities.

## Screenshots

TBD

## Additional Context

Selection is moved with values based on markdown formatting and `linkUrl` length, I would've preferred precise selections by parsing the resulting markdown but it's a serious monster compared to using values we already know.

## Checklist

**Are your changes backward compatible? Please answer below:**

_For example, a change is not backward compatible if you removed a GraphQL field or dropped a database column._
Yes
**On a scale of 1-10 how well and how have you QA'd this change and any features it might affect? Please answer below:**
6, all link operations passed QA, missing possible edge cases tests

**For frontend changes: Tested on mobile, light and dark mode? Please answer below:**
n/a

**Did you introduce any new environment variables? If so, call them out explicitly here:**
n/a

**Did you use AI for this? If so, how much did it assist you?**
Yes, only redundancies detection

---

> [!NOTE]
> **Medium Risk**
> Touches core editor link toggling logic (including markdown-mode transformations) and adds custom selection-movement behavior, which could introduce subtle selection or formatting regressions during editing.
> 
> **Overview**
> Improves link toggling behavior in the Lexical editor, especially in *markdown mode*, by centralizing decisions in `lib/lexical/commands/links.js` (detect existing markdown links, derive default/URL-from-selection, and handle item-mention selections) rather than embedding URL/markdown heuristics in the transformer bridge.
> 
> After applying a markdown link via the transformer bridge, the command now repositions the selection to highlight the link text or the URL placeholder using the new `$moveSelection` utility. Adds `hasMarkdownLink` in `lib/md.js` and simplifies `TransformerBridgePlugin` link handling to always delegate to `$toggleLink` with the provided transformation.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit f43d90425421e37f97e355a1b3d5136a98acd068. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: lib/lexical/commands/links.js [hunk 1]
```diff
@@ -45,18 +63,52 @@ export function $toggleLink (editor, url) {
...
+
+  // XXX: arbitrary selection logic,
+  // it's the fastest way to get a backwards selection right, also considering we own the pipeline
+  if (linkUrl !== DEFAULT_URL) {
+    // select the alt text inside the square brackets
+    $moveSelection(
+      updatedSelection,
+      -(linkUrl.length + linkUrl.length + 3),
+      -(linkUrl.length + 3)
+    )
```
[the remark this answers is not in the document]
Comment (Soxasora) [reply lib/lexical/commands/links.js:100]: It's not like it works by `coincidence`, we want to move selection only when we're toggling a link **on**.

`banana`
becomes
`[banana](https://)`
and `https://` is selected

`https://google.com`
becomes
`[https://google.com](https://google.com)`
and the alt text is selected

Note that the URL is used as alt text when we're toggling a link on a URL.
These are two distinct scenarios. The first allows the user to insert a URL into a piece of selected text, which will be used as the alt text. The second enables the user to insert an alt text into a URL that they have selected.

**Comment by huumn** [comment]
QA is fine in the handful of cases I tried. Selection and insertion are areas we can iterate on overtime.
